ruby/test/CMakeLists.txt

```diff
@@ -3,39 +3,57 @@ if(BUILD_TESTING)
   # find all tests
   file(GLOB RUBY_TEST_SRC "./*.rb")
 
-  # TODO: It doesn't not work with this executable that's in build/Ruby-install/bin
-  # message("RUBY_EXECUTABLE=${RUBY_EXECUTABLE}")
+
+  # **NOTE**: we do not want to grab the conan one, which is statically built on all platforms, and is a msvc build on windows
+  # Instead, we want to grab the regularly installed one on your system
 
   set(_RUBY_POSSIBLE_EXECUTABLE_NAMES
     ruby
-    ruby2.7
-    ruby2.7.2
-    ruby27
-    ruby272)
+    ruby3.2
+    ruby3.2.2
+    ruby32
+    ruby322
+  )
 
-  # TODO: this isn't great but I haven't found a better way to locate the system ruby (and avoid the one in build/Ruby-install/)
+  # TODO: this isn't great but I haven't found a better way to locate the system ruby (and avoid the conan one)
   find_program(SYSTEM_RUBY_EXECUTABLE NAMES ${_RUBY_POSSIBLE_EXECUTABLE_NAMES}
     HINTS
-      "/usr/local/rvm/rubies/ruby-2.7.2/bin/"
-      "/usr/local/rvm/rubies/ruby-2.7.2/bin/ruby"
-      "/usr/share/rvm/rubies/ruby-2.7.2/bin/"
-      "$ENV{HOME}/.rvm/rubies/ruby-2.7.2/bin/"
+      "/usr/local/rvm/rubies/ruby-3.2.2/bin/"
+      "/usr/local/rvm/rubies/ruby-3.2.2/bin/ruby"
+      "/usr/share/rvm/rubies/ruby-3.2.2/bin/"
+      "$ENV{HOME}/.rvm/rubies/ruby-3.2.2/bin/"
 
-      "C:/Ruby27-x64/bin/"
+      "C:/Ruby32-x64/bin/"
 
-       "/usr/local/ruby272/bin/"
-       "/usr/local/ruby27/bin/"
+       "/usr/local/ruby322/bin/"
+       "/usr/local/ruby32/bin/"
```

Comment:
I realized CMake was happily picking my rvm ruby 2.7 because of this, so update the hints (and code comments)